src/permission/fs_permission.cc

```diff
@@ -168,7 +183,15 @@ bool FSPermission::RadixTree::Lookup(const std::string_view& s,
   }
 
   unsigned int parent_node_prefix_len = current_node->prefix.length();
-  const std::string path(s);
+  std::string path(s);
+  if (!case_sensitive_) {
+    std::transform(
+        path.begin(),
+        path.end(),
+        path.begin(),
+        [](unsigned char c) -> unsigned char { return std::tolower(c); });
```

Comment:
cc @bnoordhuis, I am not sure if this is related to the concern in https://github.com/nodejs/node/issues/47105#issuecomment-1477651907.